subsys/bluetooth/common/Kconfig

```diff
@@ -183,12 +191,46 @@ config BT_BUF_CMD_TX_SIZE
 	  to the maximum of 255.
 
 config BT_BUF_CMD_TX_COUNT
-	int "Number of HCI command buffers"
-	default 2
-	range 2 64
+	# This option is only visible for a user when Host and Controller are build together, or for
+	# Host-only builds.
+	int "Number of HCI command buffers" if !BT_HCI_RAW || !HAS_BT_CTLR
+	# This option is present when Host and Controller are build together, or
+	# for Host only builds, or when Controller-to-Host ACL data flow control
+	# is disabled.
+	depends on !BT_HCI_RAW || !HAS_BT_CTLR || !BT_HCI_ACL_FLOW_CONTROL
+	# BT_MESH when compiled with BT_MESH_WORKQ_SYS is starting Extended Advertising in the callback
+	# run in the system work queue and as the Host also uses the system work queue for Tx towards
+	# the Controller, one additional HCI command buffer is needed.
+	range 2 64 if BT_MESH_WORKQ_SYS
+	range 1 64
+	default 2 if BT_MESH
+	default 1
```

Comment:
@PavelVPV  I took your suggestion; Looks like `BT_MESH` always requires atleast `2`, some of the tests failed when using `1` when `BT_MESH` and `!BT_MESH_WORKQ_SYS` ... is this correct?

Reply:
Ah, I see. I checked this. While some test explicitly require 2 cmd buffers because they do 2 things in parallel (start advertising and scanning), `pb_remote_reprovision` failed because this problem with tx thread removal in the host hit mesh again. This time it is because the host's tx processor that runs on sysworkq is blocked by [the mesh node reset](https://github.com/zephyrproject-rtos/zephyr/blob/a9b76dcb1ba509f58041ca49542839021b8e6ec7/subsys/bluetooth/mesh/cfg_srv.c#L43-L46) functionality. I could fix this specific issue, but we have suspend/resume API that can also be called from sysworkq thus causing this deadlock anyway. I'll follow up on this issue separately. So for now, keep `if BT_MESH` for `default` and `range`.